x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/processors_tree/components/tree_node.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FunctionComponent, useMemo } from 'react';
+import classNames from 'classnames';
+import { i18n } from '@kbn/i18n';
+import { EuiPanel, EuiText } from '@elastic/eui';
+
+import { ProcessorInternal } from '../../../types';
+
+import { ProcessorInfo, OnActionHandler } from '../processors_tree';
+
+import { PipelineProcessorsEditorItem, Handlers } from '../../pipeline_processors_editor_item';
+import { AddProcessorButton } from '../../add_processor_button';
+
+import { PrivateTree } from './private_tree';
+
+export interface Props {
+  processor: ProcessorInternal;
+  processorInfo: ProcessorInfo;
+  onAction: OnActionHandler;
+  level: number;
+  movingProcessor?: ProcessorInfo;
+}
+
+export const TreeNode: FunctionComponent<Props> = ({
+  processor,
+  processorInfo,
+  onAction,
+  movingProcessor,
+  level,
+}) => {
+  const stringSelector = processorInfo.selector.join('.');
+  const handlers = useMemo((): Handlers => {
+    return {
+      onMove: () => {
+        onAction({ type: 'selectToMove', payload: { info: processorInfo } });
+      },
+      onCancelMove: () => {
+        onAction({ type: 'cancelMove' });
+      },
+    };
+  }, [onAction, stringSelector, processor]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  const selected = movingProcessor?.id === processor.id;
+
+  const panelClasses = classNames({
+    'pipelineProcessorsEditor__tree__item--selected': selected,
+  });
+
+  const renderOnFailureHandlersTree = () => {
+    if (!processor.onFailure?.length) {
+      return;
+    }
+    const onFailureHandlerLabelClasses = classNames({
+      'pipelineProcessorsEditor__tree__onFailureHandlerLabel--withDropZone': movingProcessor
+        ? movingProcessor.id !== processor.onFailure[0].id && movingProcessor.id !== processor.id
+        : false,
```

Comment:
You could drop the ternary here, maybe?

```suggestion
      'pipelineProcessorsEditor__tree__onFailureHandlerLabel--withDropZone':
        movingProcessor != null &&
        movingProcessor.id !== processor.onFailure[0].id &&
        movingProcessor.id !== processor.id,
```